scripts/watch_dog.py

```diff
@@ -0,0 +1,124 @@
+This will allow the mycroft user to ssh without password.  The following
+commands can be used set this up.
+    literal blocks::
+
+        $ ssh-keygen
+        $ ssh-copy-id pi@localhost
+
+Note these commands should be ran as the mycroft user.  The keygen command
+can be used just with defaults.  If you include a passcode then it will
+be required and cause the script not to work.  The second command the copy
+will copy your public key into the authorized_keys. This will require the
+password for pi before copying.  You then follow the instruction to from
+the command to check if it worked.  You should be fine.
+
+The process is as follows
+    1. Check For Changes in mic_level file
+    2. Once the changes stop for 20 sec
+        A. Kill and Restart pulseaudio
+        B. Kill ans restart mycroft
+        C. Check the mic file is changing up to 1 min.
+    3. Return to step one.
+
+"""
+import os.path
+import subprocess
+import time
+
+forever = True
+
+filename = "/tmp/mycroft/ipc/mic_level"
+"""string: location of the mic level file"""
+
+pid_files = ['service','skills','voice']
+#start_mycroft = []
+
+for i, filex in enumerate(pid_files):
+    pid_files[i] = "/tmp/mycroft/"+filex+".pid"
+    #start_mycroft.append(["./start.sh",filex])
+"""Expand the the pid_files to the full path of the pid files"""
+
+
+#start_mycroft = [["./mycroft.sh","start","-v"]]
+
+#start_mycroft = []
+
+#for startx in start_mycroft:
+#    subprocess.call(startx)
+
+oldData = ""
+
+time_idle = 0
+
+if os.path.isfile(filename):
+    oldData = open(filename).read()
+    indata = oldData
+    #while(indata==oldData):
+    #    time.sleep(1)
+    #    print "Waiting for File Changes"
+    #    indata = open(filename).read()
+"""setup for no change of mic file"""
+
+while forever:
+    if os.path.isfile(filename):
+        time.sleep(1)
+        indata = open(filename).read()
+        if (indata == oldData):
+            print("File Not changed")
+            time_idle = time_idle + 1
+            if (time_idle > 20):
+                """restart pulseaudio and set source to 0"""
+                subprocess.call(["pulseaudio","-k"])
+                subprocess.call(["pulseaudio","-D"])
+                subprocess.call(["pacmd","set-default-source","0"])
+                """kill mycroft using pid files."""
+                for filen in pid_files:
+                    pid = int(open(filen).read())
+                    print "Kill -9",pid
+                    subprocess.call(["kill","-9",str(pid)])
+                """call mycroft restart script as user pi.
+
+                Note: This is a redundent kill process so should consider
+                    removing the kill above and just use the script.
+                """
+                subprocess.call(["ssh","pi@localhost","./restart_mycroft.sh"])
+                #subprocess.call(["ssh","pi@localhost","sudo","systemctl","start","mycroft-messagebus"])
```

Comment:
Why are all these still in the code?  If they have different valid purposes, add a comments to explain when they should be used.